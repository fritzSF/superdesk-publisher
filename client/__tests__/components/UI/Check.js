import React from "react";
import Check from "../../../components/UI/Check";
import { render, fireEvent, getByText } from "@testing-library/react";

describe("UI/Check", () => {
  it("renders correctly", () => {
    const { container, getByText } = render(
      <Check label="testlabel" onClick={jest.fn()} isChecked={true} />
    );

    expect(getByText("testlabel")).toBeInTheDocument();
    expect(container.querySelector(".sd-checkbox")).toHaveClass("checked");
    expect(container.firstChild).toMatchSnapshot();
  });

  it("onClick function fired", () => {
    const onClick = jest.fn();
    const { container } = render(
      <Check label="testlabel" onClick={onClick} isChecked={true} />
    );
    const button = container.querySelector(".sd-check__wrapper");

    fireEvent.click(button);
    expect(onClick).toHaveBeenCalled();
  });
});
