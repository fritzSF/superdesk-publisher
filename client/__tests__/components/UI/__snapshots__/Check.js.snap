// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Check renders correctly 1`] = `
<span
  class="sd-check__wrapper"
>
  <span
    class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio checked"
  >
    <label>
      testlabel
    </label>
  </span>
</span>
`;
